File: java_projects/glue_code/commons-graph/src/main/java/org/apache/commons/graph/coloring/ColoredVertices.java
package org.apache.commons.graph.coloring;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import org.apache.commons.graph.ContextInitializer;
import org.graalvm.polyglot.Value;

public final class ColoredVertices<V, C> {
  private final List<C> usedColor = new ArrayList<C>();
  private final Map<V, C> coloredVertices = new HashMap<V, C>();
  private static Value clz =
      ContextInitializer.getPythonClass("/coloring/ColoredVertices.py", "ColoredVertices");
  private Value obj;

  public ColoredVertices(Value obj) {
    this.obj = obj;
  }

  public Value getPythonObject() {
    return obj;
  }

  public int getRequiredColors() {
    //
    // return usedColor.size();
    //

    // TODO: Check the type mapping below!
    return obj.invokeMember("getRequiredColors").as(int.class);
  }

  public C getColor(V v) {
    //
    // v = checkNotNull(v, "Impossible to get the color for a null Vertex");
    //
    // return coloredVertices.get(v);
    //

    // TODO: Check the type mapping below!
    return obj.invokeMember("getColor", v).as(C.class);
  }

  public boolean containsColoredVertex(V vertex) {
    //
    // return coloredVertices.containsKey(vertex);
    //

    // TODO: Check the type mapping below!
    return obj.invokeMember("containsColoredVertex", vertex).as(boolean.class);
  }

  void removeColor(V v) {
    //
    // C color = coloredVertices.remove(v);
    // usedColor.remove(color);
    //

    obj.invokeMember("removeColor", v);
  }

  void addColor(V v, C color) {
    //
    // coloredVertices.put(v, color);
    // int idx = usedColor.indexOf(color);
    // if (idx == -1) {
    // usedColor.add(color);
    // } else {
    // usedColor.set(idx, color);
    // }
    //

    obj.invokeMember("addColor", v, color);
  }

  ColoredVertices() {
    //

    this.obj = clz.invokeMember("__init__");
  }
}

File: data/recomposed_projects-manual-old/deepseek-coder-33b-instruct/body/commons-graph/src/main/org/apache/commons/graph/coloring/ColoredVertices.py
from __future__ import annotations
import re
import os
import io
import typing
from typing import *
from src.main.org.apache.commons.graph.utils.Assertions import *


class ColoredVertices:

    __usedColor: typing.List[typing.Any] = []
    __coloredVertices: Dict[V, C] = {}

    def getRequiredColors(self) -> int:
        return len(self.__usedColor)

    def getColor(self, v: typing.Any) -> typing.Any:

        v = Assertions.checkNotNull(v, "Impossible to get the color for a null Vertex")

        return self.__coloredVertices.get(v)

    def containsColoredVertex(self, vertex: typing.Any) -> bool:
        return vertex in self.__coloredVertices

    def removeColor(self, v: typing.Any) -> None:

        color = self.__coloredVertices.pop(v, None)
        if color is not None:
            self.__usedColor.remove(color)

    def addColor(self, v: typing.Any, color: typing.Any) -> None:

        self.__coloredVertices[v] = color
        idx = self.__usedColor.index(color) if color in self.__usedColor else -1
        if idx == -1:
            self.__usedColor.append(color)
        else:
            self.__usedColor[idx] = color

    def __init__(self) -> None:
        self.__usedColor = []
        self.__coloredVertices = {}
